sklearn/feature_selection/_univariate_selection.py

```diff
@@ -300,14 +295,94 @@ def f_regression(X, y, *, center=True):
     corr = safe_sparse_dot(y, X)
     corr /= X_norms
     corr /= np.linalg.norm(y)
+    return corr
 
-    # convert to p-value
+
+@_deprecate_positional_args
+def f_regression(X, y, *, center=True):
+    """Univariate linear regression tests returning F-statistic and p-values.
+
+    Quick linear model for testing the effect of a single regressor,
+    sequentially for many regressors.
+
+    This is done in 2 steps:
+
+    1. The cross correlation between each regressor and the target is computed,
+       that is, ((X[:, i] - mean(X[:, i])) * (y - mean_y)) / (std(X[:, i]) *
+       std(y)) using r_regression function.
+    2. It is converted to an F score and then to a p-value.
+
+    If p-values are not needed, r_regression can be used to rank features as
+    a slightly cheaper alternative to f_regression. Note however that
+    contrary to f_regression, r_regression values lie in [-1, 1] and can thus
+    be negative.
+
+    Alternatively, `abs_r_regression` can be used to rank features by
+    correlation magnitude instead.
+
+    Read more in the :ref:`User Guide <univariate_feature_selection>`.
+
+    Parameters
+    ----------
+    X : {array-like, sparse matrix}  shape = (n_samples, n_features)
+        The set of regressors that will be tested sequentially.
+
+    y : array of shape(n_samples).
+        The data matrix
+
+    center : True, bool,
+        If true, X and y will be centered.
+
+    Returns
+    -------
+    F : array, shape=(n_features,)
+        F values of features.
+
+    pval : array, shape=(n_features,)
+        p-values of F-scores.
+
+    See Also
+    --------
+    abs_r_regression: Absolute value of Pearson R between label and features
+        for regression tasks.
+    r_regression: Pearson R between label/feature for regression tasks.
+    f_classif: ANOVA F-value between label/feature for classification tasks.
+    chi2: Chi-squared stats of non-negative features for classification tasks.
+    SelectKBest: Select features based on the k highest scores.
+    SelectFpr: Select features based on a false positive rate test.
+    SelectFdr: Select features based on an estimated false discovery rate.
+    SelectFwe: Select features based on family-wise error rate.
+    SelectPercentile: Select features based on percentile of the highest
+        scores.
+    """
+
+    # compute the correlation
+    corr = r_regression(X, y, center=center)
     degrees_of_freedom = y.size - (2 if center else 1)
+    # convert to p-value
     F = corr ** 2 / (1 - corr ** 2) * degrees_of_freedom
     pv = stats.f.sf(F, 1, degrees_of_freedom)
     return F, pv
 
 
+def abs_r_regression(X, y, center=True):
+    """Absolute value of Pearson R from univariate linear regressions.
+
+    .. versionadded:: 1.0
+
+    This convenience wrapper is to be used with SelectKBest and other models
+    that require a statistic which is increased with significance of
+    association.
```

Comment:
```suggestion
    This wrapper is convenient when using :class:`~sklearn.feature_selection.SelectKBest`
    and other models that require a statistic which is increased with significance of
    association.
```

I am unsure of the meaning of `significance of association` thought (it might be right ;))

Reply:
> I am unsure of the meaning of significance of association thought (it might be right ;))

So do I, to be honest.